### PR TITLE
My hw jv fruit shop

### PR DESCRIPTION
**Pull request by VolodymyrKachan**

### CODE REVIEW

**Review by Mihael0204 — APPROVED** [review]

**Review by Mihael0204 — APPROVED** [review]

File: src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+public class ReaderServiceImpl implements ReaderService {
+    @Override
+    public List<String> readFromFile(String filePath) {
+        File file = new File(filePath);
+        try {
+            return Files.readAllLines(file.toPath());
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java:14]:
```suggestion
        try {
            return Files.readAllLines(Path.of(filePath));
```

File: src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    @Override
+    public String create(Map<Fruit, Integer> storage) {
+        List<String> reportList = new ArrayList<>();
+        reportList.add(HEAD_REPORT);
+        for (Map.Entry<Fruit, Integer> entry : storage.entrySet()) {
+            reportList.add(entry.getKey().getName() + "," + entry.getValue());
+        }
+        return reportList.stream().collect(Collectors.joining(System.lineSeparator()));
+    }
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java:21]: concatenate using `StringBuilder`

File: src/main/java/core/basesyntax/strategy/impl/PurchaseOperationHandler.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+        if (currentQuantity >= transaction.getCount()) {
+            Storage.storage.put(transaction.getFruit(), currentQuantity - transaction.getCount());
+        }
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/strategy/impl/PurchaseOperationHandler.java:13]: let's throw expception if we don't have enough fruits

File: src/main/java/core/basesyntax/service/impl/ParsedStringInFileImpl.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ParsedStringInFile;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ParsedStringInFileImpl implements ParsedStringInFile {
+    private static final String CHAR_FOR_SPLIT = ",";
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/service/impl/ParsedStringInFileImpl.java:10]:
```suggestion
    private static final String COMMA = ",";
```

File: src/main/java/core/basesyntax/service/impl/ParsedStringInFileImpl.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int COUNT_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> parse(List<String> fromFile) {
+        fromFile.remove(0);
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/service/impl/ParsedStringInFileImpl.java:17]:
```suggestion
        data.remove(HEADER);
```

File: src/main/java/core/basesyntax/service/ReportService.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.Fruit;
+import java.util.Map;
+
+public interface ReportService {
+    String create(Map<Fruit, Integer> storage);
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/service/ReportService.java:7]: accept entry set here

File: src/main/java/core/basesyntax/model/FruitTransaction.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public String toString() {
+        return "Transaction{"
+                + "operation='" + operation + '\''
+                + ", fruit=" + fruit
+                + ", count=" + count
+                + '}';
+    }
+
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/model/FruitTransaction.java:66]: empty line

**Review by dSkarzh — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java (hunk 1 above)
Comment (dSkarzh) [inline src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java:14]: i left you comment about this
did you fix it?

File: src/main/java/core/basesyntax/service/impl/ParsedStringInFileImpl.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ParsedStringInFile;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ParsedStringInFileImpl implements ParsedStringInFile {
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/service/impl/ParsedStringInFileImpl.java:9]:
```suggestion
public class ParserServiceImpl implements ParserService {
```

**Review by dSkarzh — APPROVED** [review]

**Pull request closed without merging**